src/cmd/packages.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The ChromiumOS Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use anyhow::anyhow;
+use anyhow::Result;
+use argh::FromArgs;
+use glob::Pattern;
+use lium::cache::KvCache;
+use lium::chroot::Chroot;
+use lium::repo::get_repo_dir;
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Tast test wrapper
+#[argh(subcommand, name = "packages")]
+pub struct Args {
+    #[argh(subcommand)]
+    nested: SubCommand,
+}
+
+pub static PACKAGE_CACHE: KvCache<Vec<String>> = KvCache::new("package_cache");
+static DEFAULT_BOARD: &str = "host";
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+enum SubCommand {
+    List(ArgsList),
+}
+pub fn run(args: &Args) -> Result<()> {
+    match &args.nested {
+        SubCommand::List(args) => run_packages_list(args),
+    }
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Get package list for the target board
+#[argh(subcommand, name = "list")]
+pub struct ArgsList {
+    /// target cros repo directory
+    #[argh(option)]
+    repo: Option<String>,
+
+    /// target board (default: host)
+    #[argh(option)]
+    board: Option<String>,
+
+    /// glob pattern of the packages
+    #[argh(positional)]
+    packages: Option<String>,
+}
+
+fn print_cached_packages(filter: &Pattern, board: &str) -> Result<()> {
+    if let Ok(Some(packages)) = PACKAGE_CACHE.get(board) {
+        for t in &packages {
+            if filter.matches(t) {
+                println!("{t}");
+            }
+        }
+        return Ok(());
+    }
+    Err(anyhow!("No cache found"))
+}
+
+fn update_cached_packages(repodir: &str, board: &str) -> Result<()> {
+    let boardopt = if board == "host" {
+        "--host".to_string()
+    } else {
+        format!("--build-target={}", board)
+    };
+    let chroot = Chroot::new(repodir)?;
+    let list = chroot.exec_in_chroot(&["cros", "workon", &boardopt, "list", "--all"])?;
+    let packages: Vec<String> = list.lines().map(|s| s.to_string()).collect::<Vec<_>>();
+    PACKAGE_CACHE.set(board, packages)?;
+    Ok(())
+}
+
+fn run_packages_list(args: &ArgsList) -> Result<()> {
+    let filter = if let Some(_packages) = &args.packages {
+        Pattern::new(_packages)?
+    } else {
+        Pattern::new("*")?
+    };
+    let board = if let Some(_board) = &args.board {
+        _board
+    } else {
+        DEFAULT_BOARD
+    };
+
+    if print_cached_packages(&filter, board).is_ok() {
+        return Ok(());
+    }
```

Comment:
This is more like a question. If a user doesn't give any additional argument to `list`, then I suppose this command should print all the packages for this board. If so, wouldn't it be safer to update the cache every time for that case?

Reply:
@mhiramat As we discussed offline, it may be a good practice to add '--cached' option to explicitly disabling the auto update of the list (for command line completion etc...) and do auto update of the list automatically by default. Could you update the code to do so? Thank you!